network/src/inbound/inbound.rs

```diff
@@ -192,32 +187,31 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
                 }
             };
 
-            // Handle Ping/Pong messages immediately in order not to skew latency calculation.
-            match &message.payload {
-                Payload::Ping(..) => {
-                    self.outbound
-                        .send_request(Message::new(Direction::Outbound(reader.addr), Payload::Pong))
-                        .await;
-                }
-                Payload::Pong => {
-                    self.peer_book.received_pong(reader.addr);
+            // Route the message to the inbound handler of this node.
+            {
+                // Handle Ping/Pong messages immediately in order not to skew latency calculation.
+                match &message.payload {
+                    Payload::Ping(block_height) => {
+                        self.outbound
+                            .send_request(Message::new(Direction::Outbound(reader.addr), Payload::Pong))
+                            .await;
+                        self.peer_book.received_ping(reader.addr, *block_height);
```

Comment:
this operation should still be queued with all other requests (i.e. handled at `process_incoming_messages`) in order to avoid being maliciously spammed with `Ping` messages; `received_pong` is guarded against this with `quality.expecting_pong`.

Reply:
Agreed, will do!